src/v/redpanda/application.cc

```diff
@@ -884,6 +886,26 @@ void application::check_environment() {
               strict_data_dir_file));
         }
     }
+
+    if (config::node().fips_mode()) {
+        if (!ss::file_exists(fips_enabled_file).get()) {
+            throw std::runtime_error(fmt::format(
+              "File '{}' does not exist.  Redpanda cannot start in FIPS mode",
+              fips_enabled_file));
+        }
+
+        auto fd = ss::file_desc::open(fips_enabled_file.data(), O_RDONLY);
+        char buf[1];
+        fd.read(buf, 1);
+        if (buf[0] != '1') {
+            throw std::runtime_error(fmt::format(
+              "File '{}' not reporting '1': '{}'.  Redpanda cannot start in "
+              "FIPS mode",
+              fips_enabled_file,
+              std::string(&buf[0], 1)));
```

Comment:
@michael-redpanda The byte you are printing may not be printable (e.g. it could be `\n` or whatever). Consider printing it in hex format?